src/graphql/account.rs

```diff
@@ -420,6 +400,96 @@ impl AccountMutation {
     }
 }
 
+async fn common_sign_in_logic(
```

Comment:
This method doesn't seem to behave consistently. Its behavior depends on `is_update_password`, which is the only differing argument between the two callers. In other words, this method is not the common part between the two callers. When we extract the common part, it should be identical across all participants.

Reply:
If there were any reason to keep this, it should have at least a different name rather than "common".

Reply:
What do you think of the function name below?
```
sign_in_logic_by_update_password
```

Reply:
It would be better to explicitly outline the process in both methods, `sign_in` and `sign_in_with_new_password`. 

I suggest the following pseudocode:

`sign_in`
    - Check the password
    - Check if there was no previous sign-in
    - Check the IP address
    - Check the number of parallel sessions
    - Sign in

`sign_in_with_new_password`
    - Check the password
    - Check the IP address
    - Check the number of parallel sessions
    - Change the password
    - Sign in

While revealing this skeleton in both methods, we need to extract the common parts as separate methods.

Reply:
I want to point out we need to check the password first. The information about whether the user is signing in for the first time or again, whether the number of sessions exceeds the limit, or whether the IP address is valid, is so confidential that it should not be accessible to anyone who has not provided the correct password. The original code was not properly implemented in this regard.

Reply:
> While revealing this skeleton in both methods, we need to extract the common parts as separate methods.

The common part might involve validating the IP address and the number of sessions.

Reply:
I considered splitting the function because it is responsible for so many functions, but there is a lot of overlap, so we made it into one function.
As you mentioned above, we will refactor the checking part by splitting it into functions and calling them.